ci.sh

```diff
@@ -122,6 +122,15 @@ fi
 echo "Unit, doc and integration tests"
 RUST_BACKTRACE=1 cargo test $FEATURES
 
+# Run tests for Parsec features that require a nightly version of the compiler
+# to function. For example: Unix peer credentials authenticator is currently
+# only functional if we use the nightly compiler, since it depends on the
+# `peer_cred` function.
+# TODO: change this list as features trickle into the stable branch.
+echo "Nightly features"
+NIGHTLY_FEATURES="--features=peer-credentials-authenticator"
+RUST_BACKTRACE=1 cargo test +nightly $NIGHTLY_FEATURES
+
```

Comment:
As the changes to add this test are fairly small, I propose to add it directly in the other PR so that we can reload the CI once the feature is in `rustc` 😃 

In the CI logs I can see:
```
error[E0554]: `#![feature]` may not be used on the stable release channel
```
are you sure the nightly version is installed? Or maybe this is because the `+nightly` needs to be before `test`?

Also last thing, I think this is fine to just have a section here specific to test the peer credentials authenticator:

```
echo "Unit testing peer credential authenticator"
RUST_BACKTRACE=1 cargo +nightly test --features=peer-credentials-authenticator
```